cscs-checks/libraries/io/hdf5_compile_run.py

```diff
@@ -71,23 +67,12 @@ def __init__(self, lang, linkage, **kwargs):
 
         self.num_tasks = 1
         self.num_tasks_per_node = 1
+        self.build_system = 'SingleSource'
+        flags = ['-%s' % linkage]
+        self.build_system.cflags = flags
+        self.build_system.cxxflags = flags
+        self.build_system.fflags = flags
```

Comment:
The linkage is part of the `ldflags`. You should simply set `self.build_systems.ldflags`.